app/src/main/java/org/literacyapp/authentication/AuthenticationActivity.java

```diff
@@ -0,0 +1,179 @@
+
+        animalOverlayHelper = new AnimalOverlayHelper(getApplicationContext());
+
+        // Initialize DB Session
+        LiteracyApplication literacyApplication = (LiteracyApplication) getApplicationContext();
+        DaoSession daoSession = literacyApplication.getDaoSession();
+        studentImageCollectionEventDao = daoSession.getStudentImageCollectionEventDao();
+    }
+
+    @Override
+    public void onCameraViewStarted(int width, int height) {
+
+    }
+
+    @Override
+    public void onCameraViewStopped() {
+
+    }
+
+    @Override
+    public Mat onCameraFrame(CameraBridgeViewBase.CvCameraViewFrame inputFrame) {
+        Mat imgRgba = inputFrame.rgba();
+        Mat imgCopy = new Mat();
+
+        // Store original image for face recognition
+        imgRgba.copyTo(imgCopy);
+
+        // Mirror front camera image
+        Core.flip(imgRgba,imgRgba,1);
+
+        List<Mat> images = ppF.getCroppedImage(imgCopy);
+        if (images != null && images.size() == 1){
+            Mat img = images.get(0);
+            if (img != null){
+                Rect[] faces = ppF.getFacesForRecognition();
+                if (faces != null && faces.length == 1){
+                    String svmString = getSvmString(img);
+                    String svmProbability = svm.recognizeProbability(svmString);
+                    Student student = getStudentFromProbability(svmProbability);
+                    numberOfTries++;
+                    if (student != null){
+                        new StudentUpdateHelper(getApplicationContext(), student).updateStudent();
+                        finish();
+                    } else if (numberOfTries >= NUMBER_OF_MAXIMUM_TRIES){
+                        startActivity(new Intent(getApplicationContext(), StudentImageCollectionActivity.class));
+                    }
+                }
+            }
+        }
+
+        // Add overlay
+        animalOverlayHelper.addOverlay(imgRgba);
+
+        return imgRgba;
+    }
+
+    @Override
+    public void onResume()
+    {
+        super.onResume();
+        TrainingHelper trainingHelper = new TrainingHelper(getApplicationContext());
+        svm = trainingHelper.getSvm();
+        tensorFlow = trainingHelper.getInitializedTensorFlow();
+        ppF = new PreProcessorFactory(getApplicationContext());
+        animalOverlayHelper.createOverlay();
+        numberOfTries = 0;
+        preview.enableView();
+    }
+
+    /**
+     * Returns the SVM string using the feature vector
+     * @param img
+     * @return
+     */
+    private synchronized String getSvmString(Mat img){
+        Mat featureVector = tensorFlow.getFeatureVector(img);
+        return svm.getSvmString(featureVector);
+    }
+
+    /**
+     * Determines if and which Student has been recognized. The Student is only returned if the probability is above 90%
+     * @param svmProbability
+     * @return
+     */
+    private synchronized Student getStudentFromProbability(String svmProbability){
```

Comment:
Exactly. I think the first case wouldn't be trivial as you said. I can set it up now. Do I add it to the existing androidTest package?